src/sync/grandpa_warp_sync.rs

```diff
@@ -198,24 +201,43 @@ impl<TSrc> GrandpaWarpSync<TSrc> {
                         virtual_machine: _,
                     },
                     _,
-                ) => return Self::Finished(Err(Error::BabeFetchEpoch(error))),
+                ) => {
+                    return (
+                        Self::InProgress(
+                            InProgressGrandpaWarpSync::warp_sync_request_from_next_source(
+                                state.sources,
+                                PreVerificationState {
+                                    start_chain_information: state.start_chain_information,
+                                },
+                                None,
+                            ),
+                        ),
+                        Some(Error::BabeFetchEpoch(error)),
+                    )
+                }
                 (babe_fetch_epoch::Query::StorageGet(storage_get), fetched_current_epoch) => {
-                    return Self::InProgress(InProgressGrandpaWarpSync::StorageGet(StorageGet {
-                        inner: storage_get,
-                        fetched_current_epoch,
-                        state,
-                    }))
+                    return (
+                        Self::InProgress(InProgressGrandpaWarpSync::StorageGet(StorageGet {
```

Comment:
`InProgressGrandpaWarpSync::warp_sync_request_from_next_source`

Reply:
I'm confused, in what circumstances would you call that here instead of returning `StorageGet`?

Reply:
Nevermind, I assumed that it was an error because it shows up in the diff